rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/ui/shaders/ShaderWidget.kt

```diff
@@ -0,0 +1,146 @@
+package com.jetbrains.rider.plugins.unity.ui.shaders
+
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.actionSystem.impl.SimpleDataContext
+import com.intellij.openapi.editor.impl.EditorImpl
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.rd.createNestedDisposable
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.wm.CustomStatusBarWidget
+import com.intellij.openapi.wm.StatusBarWidget
+import com.intellij.openapi.wm.StatusBarWidget.Multiframe
+import com.intellij.openapi.wm.impl.status.EditorBasedWidget
+import com.jetbrains.rd.platform.util.lifetime
+import com.jetbrains.rd.util.lifetime.Lifetime
+import com.jetbrains.rd.util.lifetime.SequentialLifetimes
+import com.jetbrains.rdclient.document.getFirstEditableEntityId
+import com.jetbrains.rider.UnityProjectDiscoverer
+import com.jetbrains.rider.cpp.fileType.CppFileType
+import com.jetbrains.rider.model.EditableEntityId
+import com.jetbrains.rider.model.ShaderContextData
+import com.jetbrains.rider.model.ShaderContextDataBase
+import com.jetbrains.rider.plugins.unity.UnityHost
+import icons.UnityIcons
+import java.awt.BorderLayout
+import java.awt.event.MouseEvent
+import java.awt.event.MouseListener
+import javax.swing.JComponent
+import javax.swing.JLabel
+import javax.swing.JPanel
+
+
+class ShaderWidget(project: Project) : EditorBasedWidget(project), CustomStatusBarWidget, Multiframe {
```

Comment:
How about implementing `getTooltipText()`? It would go a long way to helping understand what the popup does.  Something like "File and symbol context derived from include at {name}:{line}", or "Default file and symbol context".

Also, how about adding a help button to the header? Clicking it could take you to a wiki page on GitHub with an explanation of what's going on and why someone might want to change this.